internal/ui/termstatus/terminal_windows.go

```diff
@@ -19,6 +19,12 @@ func clearCurrentLine(wr io.Writer, fd uintptr) func(io.Writer, uintptr) {
 		return windowsClearCurrentLine
 	}
 
+	// check if the output file type is a pipe (0x0003)
+	if isPipe(fd) {
```

Comment:
Adding this check turns `return posixClearCurrentLine` into dead code. `clearCurrentLine` and `moveCursorUp` are only called when `canUpdateStatus` has returned true before. And the only happens for windows terminals or if `isPipe(fd)` returns true.

Btw, `isPipe(fd)` misses a negation which got lost in #2608. Adding that negation, would in turn make `return func(io.Writer, uintptr) {}` dead code.

The clean version of this PR would be to replace everything in `canUpdateStatus` starting from `if !isPipe(fd)` with `return false`. In that case we should also drop all other compatibility code for mintty. However, I haven't had time to test restic+mintty, so I don't know whether that feature is broken anyways or not.

Reply:
Thanks for pointing that out. I had blindly reverted the change without looking into it deeply before. I've done a deeper investigation and cleaned up the code considerably.